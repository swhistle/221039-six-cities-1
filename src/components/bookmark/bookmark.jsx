import React from "react";
import PropTypes from "prop-types";

import {withElementIsActive} from "../../hocs/withElementIsActive/withElementIsActive";

class BookmarkComponent extends React.PureComponent {
  constructor(props) {
    super(props);

    this._addToBookmarks = this._addToBookmarks.bind(this);
  }

  _addToBookmarks(hotelId, elementIsActive) {
    this.props.changeElementIsActive(elementIsActive);
    this.props.addToBookmarks(hotelId, !elementIsActive);
  }

  render() {
    const {width = 18, height = 19, elementIsActive, hotelId} = this.props;

    return <button className="place-card__bookmark-button button" type="button" onClick={() => this._addToBookmarks(hotelId, elementIsActive)}>
      <svg className={elementIsActive ? `property__bookmark-icon property__bookmark-icon--active` : `property__bookmark-icon`} width={width} height={height} viewBox="0 0 17 18" xmlns="http://www.w3.org/2000/svg">
        <path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"/>
      </svg>
      <span className="visually-hidden">To bookmarks</span>
    </button>;
  }
}

BookmarkComponent.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  addToBookmarks: PropTypes.func,
  elementIsActive: PropTypes.bool,
  changeElementIsActive: PropTypes.func,
  hotelId: PropTypes.number,
};

export {BookmarkComponent};

export default withElementIsActive(BookmarkComponent);
