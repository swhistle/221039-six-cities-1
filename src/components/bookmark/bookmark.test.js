import React from 'react';
import renderer from 'react-test-renderer';

import {BookmarkComponent} from "./bookmark.jsx";

const TEST_HOTEL_ID = 1;
const TEST_HOTEL_IS_FAVORITE = false;

const addToBookmarks = () => {};
const changeElementIsActive = () => {};

it(`BookmarkComponent correct renders`, () => {
  const bookmarkComponent = renderer
    .create(
        <BookmarkComponent hotelId={TEST_HOTEL_ID} addToBookmarks={addToBookmarks} elementIsActive={TEST_HOTEL_IS_FAVORITE} changeElementIsActive={changeElementIsActive}/>
    ).toJSON();

  expect(bookmarkComponent).toMatchSnapshot();
});
