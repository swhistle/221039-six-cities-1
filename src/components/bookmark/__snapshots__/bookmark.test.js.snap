// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkComponent correct renders 1`] = `
<button
  className="place-card__bookmark-button button"
  onClick={[Function]}
  title="add to favorites"
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height={19}
    viewBox="0 0 17 18"
    width={18}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
